// autumn-js/dist/libraries/backend/hono.mjs
import {
  secretKeyCheck
} from "./chunk-UNZHJTEY.mjs";
import {
  createRouterWithOptions
} from "./chunk-5XFVERCB.mjs";
import "./chunk-5Q7TYYPU.mjs";
import "./chunk-RZ5DRUNQ.mjs";
import "./chunk-KIW4KUB3.mjs";
import "./chunk-IBB4P4P3.mjs";
import "./chunk-SW3YG5LX.mjs";
import "./chunk-WQPFSMN6.mjs";
import {
  Autumn
} from "./chunk-CCBTGZR4.mjs";
import "./chunk-SGEUXFOF.mjs";
import {
  autumnApiUrl
} from "./chunk-KSG3E4Q2.mjs";

// src/libraries/backend/hono.ts
import { findRoute } from "rou3";
var autumnHandler = (options) => {
  const autumn = new Autumn({
    url: options.url || autumnApiUrl,
    version: options.version,
    secretKey: options.secretKey
  });
  const router = createRouterWithOptions();
  const { found, error: resError } = secretKeyCheck(options?.secretKey);
  return async (c, next) => {
    if (!found && !options.secretKey) {
      return c.json(resError, 500);
    }
    const request = new URL(c.req.url);
    const path = request.pathname;
    const searchParams = Object.fromEntries(request.searchParams);
    const match = findRoute(router, c.req.method, path);
    if (match) {
      const { data, params: pathParams } = match;
      const { handler } = data;
      let body = null;
      if (c.req.method === "POST" || c.req.method === "PUT" || c.req.method === "PATCH") {
        try {
          body = await c.req.json();
        } catch (error) {
        }
      }
      const result = await handler({
        autumn,
        body,
        path: c.req.path,
        getCustomer: async () => {
          return await options.identify(c);
        },
        pathParams,
        searchParams
      });
      return c.json(result.body, result.statusCode);
    }
    await next();
  };
};
export {
  autumnHandler
};
