// autumn-js/dist/libraries/backend/tanstack.mjs
import {
  secretKeyCheck
} from "./chunk-UNZHJTEY.mjs";
import {
  createRouterWithOptions
} from "./chunk-5XFVERCB.mjs";
import "./chunk-5Q7TYYPU.mjs";
import "./chunk-RZ5DRUNQ.mjs";
import "./chunk-KIW4KUB3.mjs";
import "./chunk-IBB4P4P3.mjs";
import "./chunk-SW3YG5LX.mjs";
import "./chunk-WQPFSMN6.mjs";
import {
  Autumn
} from "./chunk-CCBTGZR4.mjs";
import "./chunk-SGEUXFOF.mjs";
import {
  autumnApiUrl
} from "./chunk-KSG3E4Q2.mjs";

// src/libraries/backend/tanstack.ts
import { findRoute } from "rou3";
import { json } from "@tanstack/react-start";
var autumnHandler = (options) => {
  const autumn = new Autumn({
    url: autumnApiUrl,
    version: options.version
  });
  const router = createRouterWithOptions();
  let { found, error: resError } = secretKeyCheck(options?.secretKey);
  const handleRequest = async (ctx) => {
    const { request } = ctx;
    if (!found && !options.secretKey) {
      return new Response(JSON.stringify(resError), {
        status: resError.statusCode
      });
    }
    const url = new URL(request.url);
    const searchParams = Object.fromEntries(url.searchParams);
    const pathname = url.pathname;
    const method = request.method;
    const match = findRoute(router, method, pathname);
    if (!match) {
      return new Response(JSON.stringify({ error: "Not found" }), {
        status: 404,
        headers: { "Content-Type": "application/json" }
      });
    }
    const { data, params: pathParams } = match;
    const { handler } = data;
    let body = null;
    if (method === "POST" || method === "PUT" || method === "PATCH") {
      try {
        body = await request.json();
      } catch (error) {
      }
    }
    try {
      const result = await handler({
        autumn,
        body,
        path: pathname,
        getCustomer: async () => await options.identify(ctx),
        pathParams,
        searchParams
      });
      return json(result.body, { status: result.statusCode });
    } catch (error) {
      console.error("Autumn handler error:", error.message);
      return json({ error: error.message }, { status: 500 });
    }
  };
  return {
    GET: handleRequest,
    POST: handleRequest,
    PUT: handleRequest,
    PATCH: handleRequest,
    DELETE: handleRequest
  };
};
export {
  autumnHandler
};
